ncm-grub/src/main/perl/grub.pm

```diff
@@ -124,496 +140,509 @@ sub password {
     $val .= $password;
 
     $grub_fh->add_or_replace_lines(qr/^password\s+/,
-                              qr/^password $val$/,
-                              "password $val\n",
-                              $self->main_section_offset($grub_fh),
-                              );
-}
+                                   qr/^password $val$/,
+                                   "password $val\n",
+                                   $self->main_section_offset($grub_fh),
+                                   );
 
-# get the position for the main section
-# ie. after the header comments
-sub main_section_offset {
-    my ($self, $fh) = @_;
-    my ($start, $end) = $fh->get_header_positions();
-    return $start == -1 ? BEGINNING_OF_FILE : (SEEK_SET, $end);
+    return SUCCESS;
 }
 
-##########################################################################
-sub Configure {
-##########################################################################
-    my ( $self, $config ) = @_;
 
-    my $grubby = '/sbin/grubby';
-    my $prefix = '/boot';
-    if ( $config->elementExists("/software/components/grub/prefix") ) {
-        $prefix = $config->getValue("/software/components/grub/prefix");
-    }
-    my $kernelname    = 'vmlinuz';
-    my $kernelversion = '';
-    if ( $config->elementExists($PATH_KERNEL_VERSION) ) {
-        $kernelversion = $config->getValue($PATH_KERNEL_VERSION);
-    }
-    my $fulldefaultkernelpath;
+=item serial_console
 
-    # An undefined kernel version or an empty string are treated equally
-    if ( !$kernelversion ) {
-        $self->debug( 1, "No kernel version defined: default kernel will not be set" );
-    } else {
-        $fulldefaultkernelpath = $prefix . '/' . $kernelname . '-' . $kernelversion;
-    }
+Configure the grub serial console settings (C<ttyS> devices only)
+by editing the grub conf via filehandle C<grub_fh>
+(a C<CAF::FileEditor> instance, which is not closed in this method).
 
-    my $grub_fh = CAF::FileEditor->new( $GRUB_CONF,
-        owner => "root",
-        group => "root",
-        mode  => 0400,
-        log   => $self );
+Returns undef on failure, the console kernel commandline option
+(or empty string if none is to be configured) on success.
```

Comment:
the method also generates the console kernel commandline option, not sure how to combine these 2 otherwise.